src/librustc_session/options.rs

```diff
@@ -968,4 +981,6 @@ options! {DebuggingOptions, DebuggingSetter, basic_debugging_options,
         "link the `.rlink` file generated by `-Z no-link`"),
     new_llvm_pass_manager: Option<bool> = (None, parse_opt_bool, [TRACKED],
         "use new LLVM pass manager"),
+    src_file_checksum: Option<SourceFileChecksumKind> = (Some(SourceFileChecksumKind::Md5), parse_src_file_checksum, [UNTRACKED],
```

Comment:
This affects the output object files, so it should be `TRACKED`. You could also save the value of the previous compilation somewhere and then make the [codegen dep node](https://doc.rust-lang.org/nightly/nightly-rustc/rustc/mir/mono/struct.CodegenUnit.html#method.codegen_dep_node) dirty [here](https://github.com/rust-lang/rust/blob/57e1da59cd0761330b4ea8d47b16340a78eeafa9/src/librustc_codegen_llvm/base.rs#L108) if it changes.